### ISSUE TITLE
Router - add a hook for initial state

### ISSUE THREAD

**Issue by codingismy11to7** [japgolly/scalajs-react#102]
It'd be nice to have a way to hook into the Router so that the initial location can be hooked (like the `onRouteChange` hook), or having `onRouteChange` called for the initial load. Not super important, just setting the window title in my case, but don't want to have client-side code changing titles with `RoutingRules.Loc` matches during `onRouteChange` and then a different piece of code on the server operating on `/path` strings just for the initial page load.

**Comment by ramnivas** [comment]
+1 I recently faced a situation, where this would have helped. In my case, I need to do some page-level initialization (non-React) upon rendering any page--including the initial page.

Either of your suggestion would work fine for my use case. I guess calling `onRouteChange` for the initial load could be a bit controversial, since that will be akin to calling an observer even for the initial setting of values in an observable. 

Alternatively, since Router is a React component, I wonder if there is a way for users to specify `componentDidMount` (I looked through the source code but got a bit lost). I believe that will also help with your and my use case.

**Comment by codingismy11to7** [comment]
@ramnivas smart - this works fine for me:

```scala
  val routerComponent = {
    Router.componentUnbuilt(router)
      .componentDidMount(scope ⇒ setWindowTitle(scope.state))
      .buildU
  }
```

**Comment by ramnivas** [comment]
@codingismy11to7 Cool! I will try the same shortly.

**Comment by codingismy11to7** [comment]
@japgolly if that's supported (and will remain supported), maybe a line in the router docs pointing that out would be better than some new method of hooking into the initial state

**Comment by japgolly** [comment]
Good on you for looking around, I'd forgotten about that function. That
method is there exactly for these kind of situations​ (ones that I didn't
foresee). You're safe to use it, it'll be staying around. A note in the
docs would be nice; PR?

As to whether or not to add a hook, let's keep this open.

**Comment by ngbinh** [comment]
It works but feel hacky to me as we have to unbuilt then build the router.

**Comment by japgolly** [comment]
Closed via 93bbc5f1fc1fa3af7b30c7c7ef6689ed992f061b.

There's a [new Router](https://github.com/japgolly/scalajs-react/tree/93bbc5f1fc1fa3af7b30c7c7ef6689ed992f061b/extra/src/main/scala/japgolly/scalajs/react/extra/router2) coming in 0.9 that fixes this.

In `RouterConfig` there are `{set,on}PostRender` methods ([see here](https://github.com/japgolly/scalajs-react/blob/93bbc5f1fc1fa3af7b30c7c7ef6689ed992f061b/extra/src/main/scala/japgolly/scalajs/react/extra/router2/RouterConfig.scala#L32-L46)). The callback receives `(Option[Page], Page)` and for this case of an initial state hook, the `Option[Page]` will be `None`.
